src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,33 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    private Random random = new Random();
+    private Figure[]array = new Figure[5];
+
+    public Random getRandom() {
+        return random;
+    }
+
+    public Figure[] getArray() {
+        return array;
+    }
+
+    public Figure getRandomFigure() {
+        getArray() [0] = new Circle(new ColorSupplier().getRandomColor(), getRandom().nextInt(10));

```

Comment:
Use random to get a random number within the number of shape types.
Then make a switch case, where for each number create a new type of figure.